### PR TITLE
Order order cycle list #1884

### LINKED ISSUE #1884: Order "order cycle" list from most recent to oldest in "orders" filtering
We just got a feedback from a user (who actually is becoming kind of a very active tester apparently ;-)):
she got lost in trying to filter the orders by "order cycle" as she has many and they were proposed in a random order, whereas in other places on the platform they are ordered from most recent to oldest, so it's easier to search.

To do: display order cycles from most recent to oldest in "order cycle" filter field in "orders" page.

In reports for instance:
screenshot from 2017-09-27 22-22-38

Currently in orders page:
screenshot from 2017-09-27 22-22-52

### PR DESCRIPTION
**Pull request by KeirOsborn**
#1967 replaced by this. 

Closes #1884 (added by Myriam)

### CODE REVIEW

**Comment by daniellemoorhead** [comment]
Closing this as replaced by #1967

**Comment by myriamboure** [comment]
@KeirOsborn I am a bit confused, is this PR #2125 replacing the older one #1967 ? This one is from yesterday so I believe so but your message is a bit confusing as you say "#1967 replaces this". Shall we close #1967 in favor of this PR?

**Comment by myriamboure** [comment]
Tell me also @KeirOsborn if it is ready for code review again :-)

**Comment by KeirOsborn** [comment]
Ha. Sorry yes that is confusing. I guess it was meant to be "#1967 - Closes this." or something like that. Clearly I considered it was too late in the day to attempt a valid sentence.

**Comment by myriamboure** [comment]
Ok just asking some other reviewer to confirm as Pau validated before you confirmed it was ready for review :) Once this is done will move to test ready. Thanks @KeirOsborn !!!

File: app/overrides/spree/admin/orders/index/add_distributor_and_order_cycle_filter_inputs.html.haml.deface [hunk 1]
```diff
@@ -9,5 +9,5 @@
 .field-block.alpha.eight.columns
   = label_tag nil, t(:order_cycles)
   = select_tag("q[order_cycle_id_in]",
-    options_for_select(OrderCycle.managed_by(spree_current_user).map {|oc| [oc.name, oc.id]}, params[:order_cycle_ids]),
+    options_for_select(OrderCycle.managed_by(Spree::User.last).where('order_cycles.orders_close_at is not null').order('order_cycles.orders_close_at DESC').map {|oc| [oc.name, oc.id]}, params[:order_cycle_ids]),
```
Comment (enricostano) [inline app/overrides/spree/admin/orders/index/add_distributor_and_order_cycle_filter_inputs.html.haml.deface]: `Spree::User.last` will pick whatever user that is not necessarily the current user. I don't think it's a good idea.

@KeirOsborn could you please explain?

**Comment by KeirOsborn** [comment]
Oops, not sure how that got in there. That should be current_user as it was in the old version. Hang on.

**Comment by KeirOsborn** [comment]
Ok. That's better.

**Comment by pacodelaluna** [comment]
Deployed right now on french staging, https://staging.openfoodfrance.fr/

**Comment by RachL** [comment]
It's good for me ! @myriamboure here are my testing notes : https://docs.google.com/document/d/1ERO-yNfQuRXAKUF4un8Wct52or-IqsjWbS_qZQKxBbs/edit#

**Comment by myriamboure** [comment]
Good @RachL thank you, I checked and everything is all right for me, ready to go.

**Pull request merged**